### PR TITLE
[WIP] Implement job authorization

### PR DESCRIPTION
**Pull request by sbliven**
Update endpoint and instance authorization for the job controller.

This continues work on #620.

### CODE REVIEW

File: src/casl/casl-ability.factory.ts [hunk 1]
```diff
@@ -803,11 +804,13 @@ export class CaslAbilityFactory {
       can(AuthOp.JobsRead, JobClass);
       can(AuthOp.JobsCreate, JobClass);
       can(AuthOp.JobsUpdate, JobClass);
+      can(AuthOp.JobsDelete, JobClass);
```
Comment (nitrosx) [inline src/casl/casl-ability.factory.ts:807]: This should not be here.
Only if you are part of the DeleteGroups, you are allowed to delete

File: src/casl/casl-ability.factory.ts [hunk 2]
```diff
@@ -803,11 +804,13 @@ export class CaslAbilityFactory {
...
       // -------------------------------------
       // data instance authorization
       can(AuthOp.JobsReadAny, JobClass);
       can(AuthOp.JobsCreateAny, JobClass);
       can(AuthOp.JobsUpdateAny, JobClass);
+      can(AuthOp.JobsDeleteAny, JobClass);
```
Comment (nitrosx) [inline src/casl/casl-ability.factory.ts:813]: Same as above

**Review by nitrosx — CHANGES_REQUESTED** [review]
Please check how DeleteDatasets permissions are implemented and make the relevant changes to the casl-ability.factory

File: src/casl/casl-ability.factory.ts (hunk 1 above)
Comment (sbliven) [reply src/casl/casl-ability.factory.ts:807]: Oh, I thought admins had all permissions

**Comment by nitrosx** [comment]
@sbliven Delete permissions are separate. You need to specifically list the group in DELETE_GROUPS if you want that an admin group has also delete privileges.

**Comment by nitrosx** [comment]
@sbliven : 
Do you want me to work on this while I work on the instance authorization and jobs authorization in general?

File: src/casl/casl-ability.factory.ts (hunk 1 above)
Comment (bpedersen2) [reply src/casl/casl-ability.factory.ts:807]: I would also   expect admins to be  able to delete Jobs ( while the ability to delete datasets is  on a  different level)

**Comment by sbliven** [comment]
@nitrosx I don't have a good overview of what abilities should be assigned. It's probably better if I revert the casl-ability.factory.ts changes and let you implement those. I was mostly focused on the controller changes with this code.

I expect this PR will be closed and incorporated into @despadam's branch instead.

**Comment by nitrosx** [comment]
This PR has been superseded by #1173.
Closing it

**Pull request closed without merging**